lib/puppet/util/monkey_patches.rb

```diff
@@ -28,34 +28,35 @@ def daemonize
 end
 
 # (#19151) Reject all SSLv2 ciphers and handshakes
-require 'openssl'
-class OpenSSL::SSL::SSLContext
-  if DEFAULT_PARAMS[:options]
-    DEFAULT_PARAMS[:options] |= OpenSSL::SSL::OP_NO_SSLv2 | OpenSSL::SSL::OP_NO_SSLv3
-  else
-    DEFAULT_PARAMS[:options] = OpenSSL::SSL::OP_NO_SSLv2 | OpenSSL::SSL::OP_NO_SSLv3
-  end
-  if DEFAULT_PARAMS[:ciphers]
-    DEFAULT_PARAMS[:ciphers] << ':!SSLv2'
-  end
+require 'puppet/ssl/openssl_loader'
+if !defined?(JRUBY_VERSION)
```

Comment:
Should use the `Puppet::Util::Platform.jruby_fips?` introduced elsewhere in the PR